CONTRIBUTING.md

```diff
@@ -0,0 +1,139 @@
+- Limit the subject line to 50 characters
+- Capitalize the subject line
+- Wrap the body at 72 characters
+- You can also use commitizen cli to check your commit messages. [Commitizen](https://commitizen.github.io/cz-cli/)
+Reference: [how to write good commits](https://www.conventionalcommits.org/en/v1.0.0/)
+
+## Directory structure
+
+Every feature should have a separate folder in the pages directory, all the routes related to the feature should be in that folder.  
+
+```
+
+  └── __tests__
+        └── HomeTest.js
+  
+  └── components
+        ├── Sidenav.js
+        ├── Button.js
+        └── Loader.js
+  
+  └── cypress
+        └── components
+              └── home.spec.js
+        
+  └── helpers
+        ├── withAuth.js
+        ├── withoutAuth.js
+        ├── edittemplate.js
+        └── activelink.js
+        
+  └── pages
+       	├── Auth
+        |   ├── login.js
+        |   └── signup.js
+        | 
+       	├── Dashboard
+        |  	  ├── statistics.js
+        |  	  ├── paymentlogs.js
+        |  	  ├── orderlogs.js
+        |  	  ├── restaurantprofile.js
+        |  	  └── fooditems.js
+        |
+        ├── Template
+        |  	  ├── chooseTemplate.js
+        |  	  ├── templateone.js
+        |  	  └── templatetwo.js
+        |
+        ├── Restaurant
+        |  	  └── [restaurant].js (Dynamic Route)
+        |      
+        ├── End Customer
+        |     ├── login.js
+        |     ├── signup.js
+        |     └── checkkout.js
+        |
+        ├── _app.js
+        ├── _document.js
+        └── index.js
+  
+  └── public
+        └── image.png        
+
+  └── services
+        └── firebase.js
+
+  └── styles
+        └── global.js         
+
+```          	
+
+## Naming
+
+Lowercase is preferred for folder and file names, one exception to this is Files which represent the React components, for which Capitalized is preferred. [Examples for Naming Coventions](https://github.com/vercel/next.js/tree/canary/examples) 
+
+## Routing in Nextjs
+
+You can refer this link to better understand routing in nextjs. [Routing](https://nextjs.org/docs/routing/introduction)
+
+## Styles
+
+- `components` will contain reusable UI component for eg. `Button`, `Input` etc.
+- If a style is specific only to the component, and cannot be used anywhere else, use inline styling using styled-components.
+- All CSS should be used with styled-components and bootstrap. 
+
+## Testing
+
+- Write unit tests using jest in the __test__ folder.
+- Write integration test using cypress in cypress/integration folder. [How to write cypress tests](https://docs.cypress.io/guides/getting-started/writing-your-first-test#Add-a-test-file)
+- Make sure you cover at least 50% of the test-case scenario.
+- You can generate "test coverage" for modified files using the `npm run test:coverage` command.
+- You can set threshold for test coverage inside jest configuration. [Jest Coverage Threshold](https://jestjs.io/docs/configuration#coveragethreshold-object)  
```

Comment:
This is not something that we recommend changing. It should be imposed by us.

Reply:
npm run coverage:report, this will show us what is our overall coverage and give us error, if  is below 50%

Reply:
Can you document the same here

Reply:
I have added that in the latest commit